source/mqtt/Mqtt5Client.cpp

```diff
@@ -21,10 +21,23 @@ namespace Aws
     {
         namespace Mqtt5
         {
+            Mqtt5to3AdapterOptions::Mqtt5to3AdapterOptions() : m_port(0), m_overwriteWebsocket(false) {}
+
             Mqtt5Client::Mqtt5Client(const Mqtt5ClientOptions &options, Allocator *allocator) noexcept
                 : m_client_core(nullptr)
             {
                 m_client_core = Mqtt5ClientCore::NewMqtt5ClientCore(options, allocator);
+                m_mqtt5to3AdapterOptions = options.NewMqtt5to3AdapterOptions();
+            }
+
+            std::shared_ptr<Crt::Mqtt::MqttConnection> Mqtt5Client::NewConnection() noexcept
+            {
+                if (m_client_core == nullptr)
+                {
+                    AWS_LOGF_DEBUG(AWS_LS_MQTT5_CLIENT, "Failed to create mqtt3 connection: Mqtt5 Client is invalid.");
```

Comment:
I think it is a good idea to set the error level to ERROR for those "invalid mqtt5 client" issue here. As if the issue happened (the client core is `nullptr`), it usually means the Mqtt5Client is unusable, and we should expose it to user. 
Since we've been using DEBUG log for all other operations, I will keep it here for now and create a separate PR to update the log level all together.